ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilderTest.java

```diff
@@ -187,4 +189,22 @@ public void modifierOnDefinedMilestoneIsAppliedButDoesNotGetAnExtraMilestoneCrea
 
     verify(modifier, times(1)).apply(any());
   }
+
+  @Test
+  public void isOnForkBoundary() {
+    when(configOptions.getBerlinBlockNumber()).thenReturn(OptionalLong.of(1L));
+    when(configOptions.getLondonBlockNumber()).thenReturn(OptionalLong.of(2L));
+    when(configOptions.getMergeNetSplitBlockNumber()).thenReturn(OptionalLong.of(4L));
+    final HeaderBasedProtocolSchedule protocolSchedule = builder.createProtocolSchedule();
+
+    assertThat(protocolSchedule.isOnForkBoundary(header(0))).isEqualTo(true);
```

Comment:
which fork boundary is 0?

Reply:
Frontier. This is how ProtocolSchedule works, for better or worse :)
https://github.com/hyperledger/besu/blob/main/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java#L157

Note, that there's no 'default' or 0 value for the timestamp-based schedule though since we need to fall back to number-based schedule instead.